internal/host/host.go

```diff
@@ -1206,8 +1206,8 @@ func (m *Manager) GetHostByKubeKey(key types.NamespacedName) (*common.Host, erro
 	return host, nil
 }
 
-func (m *Manager) UnRegisterHost(ctx context.Context, hostID, clusterID string) error {
-	return m.db.Where("id = ? and cluster_id = ?", hostID, clusterID).Delete(&common.Host{}).Error
+func (m *Manager) UnRegisterHost(ctx context.Context, hostID, infraEnvID string) error {
+	return common.DeleteHostFromDB(m.db, hostID, infraEnvID)
```

Comment:
What about the API DeregisterHost ? The internal implementation has changed

Reply:
I have added V2Deregister.  DeregisterHost will be only used when infra env id is the same as cluster id, so no need to change it

Reply:
How will it be used by the UI if it needs to delete host?

Reply:
It will use Deregister Host with cluster ID. since cluster id is equal to infra-env id for V1, the code in host.go and common.go will have no problem in deleting it